src/ChainSafe.Gaming.Lootboxes.Chainlink/ILootboxService.cs

```diff
@@ -100,5 +120,26 @@ async Task<List<LootboxTypeInfo>> FetchAllLootboxes()
                 .Select(i => new LootboxTypeInfo { TypeId = typeIds[i], Amount = balances[i] })
                 .ToList();
         }
+
+        /// <summary>
+        /// Gets the native currency price to buy a lootbox.
+        /// </summary>
+        /// <returns>The native currency price as a string.</returns>
+        Task<BigInteger> GetPrice();
```

Comment:
What is the rarity of the lootbox that you get here?

Reply:
None, it calls the price state var sol side

```
ILootboxFactory private immutable FACTORY;
  address private immutable VIEW;
  ...
  ...
  uint private price; // Native currency needed to buy a lootbox.
```

which is then used here as a conditional in the buy function, it's not tied to a rarity afaics, the buy function just determines (amount * price) and mints, i think amount here is what is determining the rarity if its amount/rarity and not quantity.

```
function buy(uint _amount, uint _maxPrice) external payable {
    address payable sender = payable(_msgSender());
    uint currentPrice = price;
    if (currentPrice == 0 || currentPrice > _maxPrice) revert UnexpectedPrice(currentPrice);
    uint valueNeeded = _amount * currentPrice;
    if (msg.value < valueNeeded) revert InsufficientPayment();
    _mint(sender, 1, _amount, '');
    uint remainder = msg.value - valueNeeded;
    if (remainder > 0) {
      sender.sendValue(remainder);
    }
    emit Sold(sender, _amount, valueNeeded);
  }
  ```